### ISSUE TITLE
about the results

### ISSUE THREAD

**Issue by FayeXXX** [wenhaofang/Seq2SeqDiverse#1]
Thank you for sharing. 
I've run your code, but found that the train loss is high and the valid bleu score is 0.25, which is far lower. I don't know what's wrong with it. 

2023-02-15 14:50:04,811][INFO] [Epoch 0] Train Loss: 4.1881, Valid Loss: 4.0750, Valid BLEU: 0.1622
100%|██████████| 2099/2099 [03:40<00:00,  9.53it/s]
100%|██████████| 117/117 [00:04<00:00, 25.32it/s]
[2023-02-15 14:54:03,340][INFO] [Epoch 1] Train Loss: 3.2436, Valid Loss: 3.7602, Valid BLEU: 0.2062
100%|██████████| 2099/2099 [03:37<00:00,  9.64it/s]
100%|██████████| 117/117 [00:04<00:00, 27.74it/s]
[2023-02-15 14:57:51,510][INFO] [Epoch 2] Train Loss: 2.8967, Valid Loss: 3.5986, Valid BLEU: 0.2251
100%|██████████| 2099/2099 [03:36<00:00,  9.69it/s]
100%|██████████| 117/117 [00:04<00:00, 26.56it/s]
[2023-02-15 15:01:38,419][INFO] [Epoch 3] Train Loss: 2.6742, Valid Loss: 3.5157, Valid BLEU: 0.2358
100%|██████████| 2099/2099 [03:41<00:00,  9.46it/s]
100%|██████████| 117/117 [00:04<00:00, 26.77it/s]
[2023-02-15 15:05:30,682][INFO] [Epoch 4] Train Loss: 2.5142, Valid Loss: 3.4461, Valid BLEU: 0.2463
100%|██████████| 2099/2099 [03:47<00:00,  9.23it/s]
100%|██████████| 117/117 [00:04<00:00, 26.68it/s]
[2023-02-15 15:09:28,801][INFO] [Epoch 5] Train Loss: 2.3905, Valid Loss: 3.4239, Valid BLEU: 0.2536
100%|██████████| 2099/2099 [03:55<00:00,  8.92it/s]

**Comment by FayeXXX** [comment]
In misc.py line 41, there might be a mistake in the loss function. The loss should be computed from [0:-1], which is:
loss = criterion(
                outputs[:, 0:-1].reshape(-1, outputs.shape[-1]),
                targets[:, 1:].reshape(-1)
            )

**Comment by wenhaofang** [comment]
> Thank you for sharing. I've run your code, but found that the train loss is high and the valid bleu score is 0.25, which is far lower. I don't know what's wrong with it.

Thanks for your attention. The BLEU scores in most papers are based on a percentage system. The result above, 0.2536, is actually 25.36 on the percentage system. I think it's a quite reasonable score for this paraphrasing task as a baseline.

**Comment by wenhaofang** [comment]
> In misc.py line 41, there might be a mistake in the loss function. The loss should be computed from [0:-1], which is: loss = criterion( outputs[:, 0:-1].reshape(-1, outputs.shape[-1]), targets[:, 1:].reshape(-1) )

Thanks for your advise. In my opinion, the interception of `output` and `target` depends on the data processing. Maybe both are ok.
